lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -215,8 +215,7 @@ def integration_message
         def create_or_update_build_phase(target, phase_name, phase_class = Xcodeproj::Project::Object::PBXShellScriptBuildPhase)
           prefixed_phase_name = BUILD_PHASE_PREFIX + phase_name
           build_phases = target.build_phases.grep(phase_class)
-          build_phases.find { |phase| phase.name == prefixed_phase_name } ||
-            build_phases.find { |phase| phase.name == phase_name }.tap { |p| p.name = prefixed_phase_name if p } ||
+          build_phases.find { |phase| phase.name && phase.name.end_with?(phase_name) }.tap { |p| p.name = prefixed_phase_name if p } ||
```

Comment:
Shouldn't a build phase always have a non-nil `name`? Also the `if p` seems unnecessary.